primitives/transaction/src/lib.rs

```diff
@@ -135,16 +138,199 @@ impl SignatureProof {
     }
 }
 
-impl Default for SignatureProof {
+impl Default for EdDSASignatureProof {
     fn default() -> Self {
-        SignatureProof {
+        EdDSASignatureProof {
             public_key: Default::default(),
             merkle_path: Default::default(),
             signature: Signature::from_bytes(&[0u8; Signature::SIZE]).unwrap(),
         }
     }
 }
 
+bitflags! {
+    #[derive(Default, Serialize, Deserialize)]
+    /// Some authenticators may behave non-standard for signing with Webauthn:
+    ///
+    /// - they might not include the mandatory `crossOrigin` field in clientDataJSON
+    /// - they might escape the `origin`'s forward slashes with backslashes, although not necessary for UTF-8 nor JSON encoding
+    ///
+    /// To allow the WebauthnSignatureProof to construct a correct `clientDataJSON` for verification,
+    /// the proof needs to know these non-standard behaviors.
+    ///
+    /// See this tracking issue for Android Chrome: https://bugs.chromium.org/p/chromium/issues/detail?id=1233616
+    pub struct WebauthnClientDataFlags: u8 {
+        const NO_CROSSORIGIN_FIELD  = 1 << 0;
+        const ESCAPED_ORIGIN_SLASHES = 1 << 1;
+
+        // const HAS_EXTRA_FIELDS = 1 << 7; // TODO Replace client_data_extra_fields length null byte when no extra fields are present
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct WebauthnSignatureProof {
+    pub public_key: WebauthnPublicKey,
+    pub merkle_path: Blake2bMerklePath,
```

Comment:
Having the Merkle path should already allow to have an address controlled by either an EdDSA or Webauthn key. 🙂

Reply:
Yes, 1-of-x multisig works with both pubkey types :+1: